utils/generate_types/overrides-test.d.ts

```diff
@@ -245,8 +245,8 @@ export interface PlaywrightTestArgs {
   request: APIRequestContext;
 }
 
-export type PlaywrightTestProject<TestArgs = {}, WorkerArgs = {}> = Project<PlaywrightTestOptions & TestArgs, PlaywrightWorkerOptions & WorkerArgs>;
-export type PlaywrightTestConfig<TestArgs = {}, WorkerArgs = {}> = Config<PlaywrightTestOptions & TestArgs, PlaywrightWorkerOptions & WorkerArgs>;
+export type PlaywrightTestProject<TestArgs = {}, WorkerArgs = {}> = Project<PlaywrightTestOptions & TestArgs & PlaywrightTest.TestFixtures & PlaywrightTest.ProjectTestOptions, PlaywrightWorkerOptions & WorkerArgs & PlaywrightTest.WorkerFixtures & PlaywrightTest.ProjectWorkerOptions>;
+export type PlaywrightTestConfig<TestArgs = {}, WorkerArgs = {}, ProjectTestArgs = {}, ProjectWorkerArgs = {}> = Config<PlaywrightTestOptions & TestArgs & PlaywrightTest.WorkerFixtures & PlaywrightTest.ConfigTestOptions, PlaywrightWorkerOptions & WorkerArgs & PlaywrightTest.WorkerFixtures & PlaywrightTest.ConfigWorkerOptions, PlaywrightTestOptions & ProjectTestArgs & PlaywrightTest.TestFixtures & PlaywrightTest.ProjectTestOptions, PlaywrightWorkerOptions & ProjectWorkerArgs & PlaywrightTest.WorkerFixtures & PlaywrightTest.ProjectWorkerOptions>;
```

Comment:
This is _really_ gross, so hopefully there is a more idiomatic way. However, prior to adding worker-scoped type parameters via https://github.com/microsoft/playwright/pull/13952/files#diff-69583b434e25c5c454472fca02f61a2f7caa0233032e79934e6e69525c622721R64, our projects array was incorrectly types.

Prior to df15f56622913819e6eacfb14d685686f196693a, the following would fail type check:

```ts
const p: PlaywrightTestProject = { use: { vitePort: 1 } }
```

while this would pass:

```ts
const config: PlaywrightTestConfig = {
    projects: [
        { …, 
            use: { vitePort: 1 },
        } 
    ]
} 
```